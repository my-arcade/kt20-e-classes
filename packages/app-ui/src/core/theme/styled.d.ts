import 'styled-components';
declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      primary: string;
      secondary: string;
      tertiary: string;
      metaPrimary: string;
      background: string;
      text: string;
      metaText: string;
      clearText: string;
      clearBackground: string;
      dividerColor: string;

       // Monochrome
      lightest: string;
      lighter: string;
      light: string;
      mediumlight: string;
      medium: string;
      mediumdark: string;
      dark: string;
      darker: string;
      darkest: string;

      border: string;

      // Status
      success: string;
      danger: string;
      warning: string;
    },
    spacing: {
      padding: {
        small: number;
        medium: number;
        large: number;
      },
      borderRadius: {
        small: number;
        default: number;
      },
    },
    typography: {
      type: {
        primary: string;
        code: string;
      },
      weight: {
        extralight: string;
        light: string;
        regular: string;
        bold: string;
        extrabold: string;
        black: string;
      },
      size: {
        s1: number;
        s2: number;
        s3: number;
        m1: number;
        m2: number;
        m3: number;
        l1: number;
        l2: number;
        l3: number;
        code: number;
      },
    }
  }
}