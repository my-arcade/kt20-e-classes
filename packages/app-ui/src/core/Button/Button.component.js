import React, { MouseEvent } from 'react';
import styled, { DefaultTheme } from 'styled-components';

// not finished 

interface ButtonProps {
  readonly danger: boolean,
  readonly disabled: boolean,
  readonly onClick: (event: MouseEvent<HTMLElement>) => void,
  readonly type: 'hollow' | 'regular',
  readonly size: 'small' | 'regular' | 'large',
  readonly theme: DefaultTheme
}

// const defaultProps : ButtonProps = {
//   type: 'hollow',
//   size: 'regular',
//   danger: false,
//   disabled: false,
//   onClick: () => {}
// }

const ButtonStyled = styled.button<ButtonProps>`
  background-color: ${({ theme, danger, disabled }) => disabled ? theme.metaText : (danger ? theme.danger : theme.primary)};
  color: white;
  border: 1px solid transparent;
  outline: none;
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  cursor: pointer;
  border-radius: 5px;
  white-space: nowrap;
  padding: 14px;
  min-height: 48px;
  min-width: 212px;
  font-weight: bolder;
  display: flex;
  justify-content: center;
`;

// TODO add default props
// = defaultProps
const Button = ({...props} : ButtonProps) => (
  <ButtonStyled {...props} /> 
);

export default Button;
