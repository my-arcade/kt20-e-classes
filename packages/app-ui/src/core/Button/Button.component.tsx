import React, { MouseEvent } from "react";
import styled, { DefaultTheme } from "styled-components";

// not finished

export interface ButtonProps {
  readonly danger: boolean;
  readonly disabled: boolean;
  readonly onClick: (event: MouseEvent<HTMLElement>) => void;
  readonly size: string;
  readonly theme: DefaultTheme;
  readonly isLoading: boolean;
  readonly containsIcon: boolean;
  readonly appearance: string;
  readonly isLink: boolean;
  readonly ButtonWrapper: any;
  readonly children: React.ReactNode;
  readonly loadingText: string;
}

const Text = styled.span`
  display: inline-block;
  vertical-align: top;
`;

const Loading = styled.span`
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  opacity: 0;
`;

const APPEARANCES = {
  PRIMARY: "primary",
  PRIMARY_OUTLINE: "primaryOutline",
  SECONDARY: "secondary",
  SECONDARY_OUTLINE: "secondaryOutline",
  TERTIARY: "tertiary",
  OUTLINE: "outline",
};

const SIZES = {
  SMALL: "small",
  REGULAR: "regular",
  LARGE: "large",
};

const ButtonStyled = styled.button<ButtonProps>`
  border: 0;
  border-radius: 3em;
  cursor: pointer;
  display: inline-block;
  overflow: hidden;
  padding: ${(props: ButtonProps) =>
    props.size === SIZES.SMALL ? "8px 16px" : "13px 20px"};
  position: relative;
  text-align: center;
  text-decoration: none;
  vertical-align: top;
  white-space: nowrap;
  user-select: none;
  opacity: 1;
  margin: 0;
  background: transparent;

  font-size: ${(props: ButtonProps) =>
    props.size === SIZES.SMALL
      ? props.theme.typography.size.s1
      : props.theme.typography.size.s2}px;
  font-weight: ${(props: ButtonProps) =>
    props.theme.typography.weight.extrabold};
  line-height: 1;

  ${(props: ButtonProps) =>
    !props.isLoading &&
    `
      &:hover {
        box-shadow: rgba(0, 0, 0, 0.2) 0 2px 6px 0;
      }

      &:active {
        transform: translate3d(0, 0, 0);
      }

      &:focus {
        box-shadow: rgba${(props.theme.colors.primary, 0.4)} 0 1px 9px 2px;
      }

      &:focus:hover {
        box-shadow: rgba${(props.theme.colors.primary, 0.2)} 0 8px 18px 0px;
      }
    `}

  ${Text} {
    transform: scale3d(1,1,1) translate3d(0,0,0);
    opacity: 1;
  }

  ${Loading} {
    transform: translate3d(0, 100%, 0);
  }

  svg {
    height: ${(props: ButtonProps) =>
      props.size === SIZES.SMALL ? "14" : "16"}px;
    width: ${(props: ButtonProps) =>
      props.size === SIZES.SMALL ? "14" : "16"}px;
    vertical-align: top;
    margin-right: ${(props: ButtonProps) =>
      props.size === SIZES.SMALL ? "4" : "6"}px;
    margin-top: ${(props: ButtonProps) =>
      props.size === SIZES.SMALL ? "-1" : "-2"}px;
    margin-bottom: ${(props: ButtonProps) =>
      props.size === SIZES.SMALL ? "-1" : "-2"}px;

    /* Necessary for js mouse events to not glitch out when hovering on svgs */
    pointer-events: none;
  }

  ${(props: ButtonProps) =>
    props.disabled &&
    `
      cursor: not-allowed !important;
      opacity: 0.5;
      &:hover {
        transform: none;
      }
    `}

  ${(props: ButtonProps) =>
    props.disabled &&
    `
      cursor: default !important;
      pointer-events: none;
      &:hover {
        transform: none;
      }
    `}

  ${(props: ButtonProps) =>
    props.isLoading &&
    `
      cursor: progress !important;
      opacity: 0.7;

      ${Loading} {
        transform: translate3d(0, -50%, 0);
        opacity: 1;
      }

      ${Text} {
        transform: scale3d(0, 0, 1) translate3d(0, -100%, 0);
        opacity: 0;
      }

      &:hover {
        transform: none;
      }
    `}

  ${(props: ButtonProps) =>
    props.containsIcon &&
    `
      svg {
        display: block;
        margin: 0;
      }
      padding: ${props.size === SIZES.SMALL ? "7" : "12"}px;
    `}

  ${(props: ButtonProps) =>
    props.appearance === APPEARANCES.PRIMARY &&
    `
      background: ${props.theme.colors.primary};
      color: ${props.theme.colors.lightest};

      ${!props.isLoading &&
        `
          &:hover {
            background: rgba${(props.theme.colors.primary, 0.5)};
          }
          &:active {
            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;
          }
          &:focus {
            box-shadow: rgba${(props.theme.colors.primary, 0.4)} 0 1px 9px 2px;
          }
          &:focus:hover {
            box-shadow: rgba${(props.theme.colors.primary, 0.2)} 0 8px 18px 0px;
          }
        `}
    `}

  ${(props: ButtonProps) =>
    props.appearance === APPEARANCES.SECONDARY &&
    `
      background: ${props.theme.colors.secondary};
      color: ${props.theme.colors.lightest};

      ${!props.isLoading &&
        `
          &:hover {
            background: rgba${(props.theme.colors.secondary, 0.5)};
          }
          &:active {
            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;
          }
          &:focus {
            box-shadow: rgba${(props.theme.colors.secondary,
            0.4)} 0 1px 9px 2px;
          }
          &:focus:hover {
            box-shadow: rgba${(props.theme.colors.secondary,
            0.2)} 0 8px 18px 0px;
          }
        `}
    `}

  ${(props: ButtonProps) =>
    props.appearance === APPEARANCES.TERTIARY &&
    `
      background: ${props.theme.colors.tertiary};
      color: ${props.theme.colors.darkest};

      ${!props.isLoading &&
        `
          &:hover {
            background: rgba${(props.theme.colors.tertiary, 0.5)};
          }
          &:active {
            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;
          }
          &:focus {
            box-shadow: rgba${(props.theme.colors.darkest, 0.15)} 0 1px 9px 2px;
          }
          &:focus:hover {
            box-shadow: rgba${(props.theme.colors.darkest,
            0.05)} 0 8px 18px 0px;
          }
        `}
    `}

  ${(props: ButtonProps) =>
    props.appearance === APPEARANCES.OUTLINE &&
    `
      box-shadow: ${props.theme.colors.border} 0 0 0 1px inset;
      color: ${props.theme.colors.dark};
      background: transparent;

      ${!props.isLoading &&
        `
          &:hover {
            box-shadow: rgba${(props.theme.colors.border, 0.5)} 0 0 0 1px inset;
          }

          &:active {
            background: rgba${(props.theme.colors.border, 0.4)};
            box-shadow: transparent 0 0 0 1px inset;
            color: ${props.theme.colors.darkest};
          }

          &:active:focus:hover {
            ${
              /* This prevents the semi-transparent border from appearing atop the background */ ""
            }
            background: rgba${(props.theme.colors.border, 0.4)};
            box-shadow:  rgba${(props.theme.colors.darkest,
            0.15)} 0 1px 9px 2px;
          }

          &:focus {
            box-shadow: rgba${(props.theme.colors.border,
            0.15)} 0 0 0 1px inset, 
            rgba${(props.theme.colors.darkest, 0.15)} 0 1px 9px 2px;
          }
          &:focus:hover {
            box-shadow: rgba${(props.theme.colors.border,
            0.15)} 0 0 0 1px inset, 
            rgba${(props.theme.colors.darkest, 0.05)} 0 8px 18px 0px;
          }
        `};
    `};

  ${(props: ButtonProps) =>
    props.appearance === APPEARANCES.PRIMARY_OUTLINE &&
    `
      box-shadow: ${props.theme.colors.primary} 0 0 0 1px inset;
      color: ${props.theme.colors.primary};

      &:hover {
        box-shadow: ${props.theme.colors.primary} 0 0 0 1px inset;
        background: transparent;
      }

      &:active {
        background: ${props.theme.colors.primary};
        box-shadow: ${props.theme.colors.primary} 0 0 0 1px inset;
        color: ${props.theme.colors.lightest};
      }
      &:focus {
        box-shadow: ${props.theme.colors.primary} 0 0 0 1px inset, rgba${(props
      .theme.colors.primary,
    0.4)} 0 1px 9px 2px;
      }
      &:focus:hover {
        box-shadow: ${props.theme.colors.primary} 0 0 0 1px inset, rgba${(props
      .theme.colors.primary,
    0.2)} 0 8px 18px 0px;
      }
    `};

  ${(props: ButtonProps) =>
    props.appearance === APPEARANCES.SECONDARY_OUTLINE &&
    `
      box-shadow: ${props.theme.colors.secondary} 0 0 0 1px inset;
      color: ${props.theme.colors.secondary};

      &:hover {
        box-shadow: ${props.theme.colors.secondary} 0 0 0 1px inset;
        background: transparent;
      }

      &:active {
        background: ${props.theme.colors.secondary};
        box-shadow: ${props.theme.colors.secondary} 0 0 0 1px inset;
        color: ${props.theme.colors.lightest};
      }
      &:focus {
        box-shadow: ${props.theme.colors.secondary} 0 0 0 1px inset,
          rgba${(props.theme.colors.secondary, 0.4)} 0 1px 9px 2px;
      }
      &:focus:hover {
        box-shadow: ${props.theme.colors.secondary} 0 0 0 1px inset,
          rgba${(props.theme.colors.secondary, 0.2)} 0 8px 18px 0px;
      }
    `};
`;

const UnStyledButton = styled.button``;
const ButtonLink = styled.a``;

const ButtonComponentPicker = ({
  isLoading = false,
  isLink = false,
  appearance = APPEARANCES.TERTIARY,
  disabled = false,
  containsIcon = false,
  size = SIZES.REGULAR,
  ButtonWrapper = undefined,
  ...rest
}: ButtonProps) => {
  const ButtonComponent = isLink ? ButtonLink : ButtonWrapper || UnStyledButton;
  return <ButtonComponent {...rest} />;
};

export const Button = ({
  children,
  disabled,
  isLoading,
  loadingText,
  ...rest
}: ButtonProps) => {
  return (
    <ButtonStyled
      as={ButtonComponentPicker}
      disabled={disabled}
      isLoading={isLoading}
      {...rest}
    >
      <>
        <Text>{children}</Text>
        {isLoading && <Loading>{loadingText || "Loading..."}</Loading>}
      </>
    </ButtonStyled>
  );
};

export default Button;
