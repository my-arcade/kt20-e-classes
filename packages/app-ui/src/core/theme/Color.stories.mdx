import { Meta, ColorPalette, ColorItem } from '@storybook/addon-docs/blocks';
import theme  from '.';

<Meta title="Design System/Colors" />

# Colors

<ColorPalette>
  <ColorItem title="theme.colors.primary" subtitle="Lightning Yellow" colors={[theme.colors.primary]} />
  <ColorItem title="theme.colors.metaPrimary" subtitle="Buttery White" colors={[theme.colors.metaPrimary]} />
  <ColorItem title="theme.colors.background" subtitle="White Lilac" colors={[theme.colors.background]} />
  <ColorItem title="theme.colors.danger" subtitle="Bittersweet" colors={[theme.colors.danger]} />
  <ColorItem title="theme.colors.text" subtitle="Rhino" colors={[theme.colors.text]} />
  <ColorItem title="theme.colors.metaText" subtitle="Rhino shade" colors={[theme.colors.metaText]} />
  <ColorItem title="theme.colors.clearBackground" subtitle="White" colors={[theme.colors.clearBackground]} />
  <ColorItem
    title="Monochrome"
    colors={[
      theme.colors.darkest,
      theme.colors.darker,
      theme.colors.dark,
      theme.colors.mediumdark,
      theme.colors.medium,
      theme.colors.mediumlight,
      theme.colors.light,
      theme.colors.lighter,
      theme.colors.lightest,
    ]}
  />
</ColorPalette>
