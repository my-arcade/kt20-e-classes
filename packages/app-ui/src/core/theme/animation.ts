// Handy CSS animations for micro-interactions
import { keyframes } from 'styled-components';

export const easing = {
  rubber: 'cubic-bezier(0.175, 0.885, 0.335, 1.05)',
};

export const rotate360 = keyframes`
	from {
		transform: rotate(0deg);
	}
	to {
		transform: rotate(360deg);
	}
`;

export const glow = keyframes`
  0%, 100% { opacity: 1; }
  50% { opacity: .4; }
`;

export const float = keyframes`
  0% { transform: translateY(1px); }
  25% { transform: translateY(0px); }
  50% { transform: translateY(-3px); }
  100% { transform: translateY(1px); }
`;
