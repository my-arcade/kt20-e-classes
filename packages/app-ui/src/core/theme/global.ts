import { createGlobalStyle, css } from "styled-components";
import { colors, typography } from "./styles";

export const bodyStyles = css`
  font-family: ${typography.type.primary};
  font-size: ${typography.size.s3}px;
  color: ${colors.darkest};

  margin: 0;
  overflow-y: auto;
  overflow-x: hidden;

  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -webkit-tap-highlight-color: transparent;
  -webkit-overflow-scrolling: touch;

  * {
    box-sizing: border-box;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: ${typography.weight.regular};
    margin: 0;
    padding: 0;
  }

  button,
  input,
  textarea,
  select {
    outline: none;
    font-family: ${typography.type.primary};
  }

  sub,
  sup {
    font-size: 0.8em;
  }

  sub {
    bottom: -0.2em;
  }

  sup {
    top: -0.2em;
  }

  b,
  em {
    font-weight: ${typography.weight.bold};
  }

  hr {
    border: none;
    border-top: 1px solid ${colors.border};
    clear: both;
    margin-bottom: 1.25rem;
  }

  code,
  pre {
    font-family: ${typography.type.code};
    font-size: ${typography.size.s2 - 1}px;
  }

  code {
    display: inline-block;
    padding-left: 2px;
    padding-right: 2px;
    vertical-align: baseline;
    color: ${colors.secondary};
  }

  pre {
    line-height: 18px;
    padding: 11px 1rem;
    white-space: pre-wrap;
    background: rgba(0, 0, 0, 0.05);
    color: ${colors.darkest};
    border-radius: 3px;
    margin: 1rem 0;
  }
`;

const loaderStyles = css`
 /* Make clicks pass-through */
 #nprogress {
    pointer-events: none;
  }

  #nprogress .bar {
    position: fixed;
    z-index: 1031;
    top: 0;
    left: 0;
    width: 100%;
    height: 3px;
    background: ${({theme}) => theme.colors.primary};
  }

  /* Fancy blur effect */
  #nprogress .peg {
    display: block;
    position: absolute;
    right: 0;
    width: 100px;
    height: 100%;
    box-shadow: 0 0 5px #fff, 0 0 10px #fff, 0 0 15px #fff, 0 0 20px #fff;
    opacity: 1.0;
    -webkit-transform: rotate(3deg) translate(0px, -4px);
    -ms-transform: rotate(3deg) translate(0px, -4px);
    transform: rotate(3deg) translate(0px, -4px);
  }

  /* Remove these to get rid of the spinner */
  #nprogress .spinner {
    display: block;
    position: fixed;
    z-index: 1031;
    top: 15px;
    right: 15px;
  }

  #nprogress .spinner-icon {
    width: 18px;
    height: 18px;
    box-sizing: border-box;
    border: solid 2px transparent;
    border-radius: 50%;
    -webkit-animation: nprogress-spinner 400ms linear infinite;
    animation: nprogress-spinner 400ms linear infinite;
  }

  .nprogress-custom-parent {
    overflow: hidden;
    position: relative;
  }

  .nprogress-custom-parent #nprogress .spinner,
  .nprogress-custom-parent #nprogress .bar {
    position: absolute;
  }
`

// Allow access font to settings but control how and where to load the font.
export const fontUrl =
  "https://fonts.googleapis.com/css2?family=Lato:wght@400;700;900&display=swap";

export const GlobalStyle = createGlobalStyle`
  @import url('${fontUrl}');
  
  
  body {
    ${bodyStyles}
  }

  ${loaderStyles}

  .react-loading-skeleton {
      background-color: ${({theme}) => theme.colors.skeleton.base};
      background-image: linear-gradient(
        90deg,
        ${({theme}) => theme.colors.skeleton.base},
        ${({theme}) => theme.colors.skeleton.highlight},
        ${({theme}) => theme.colors.skeleton.base}
      );
    }

  html, body, #app {
    height: 100%;
    min-height: 100%;
    width: 100%;
  }
`;
