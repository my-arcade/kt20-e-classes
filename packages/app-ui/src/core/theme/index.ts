import { DefaultTheme } from "styled-components";
import { colors, typography, spacing } from "./styles"

const theme: DefaultTheme = {
  colors,
  typography,
  spacing,
};

export default theme;
