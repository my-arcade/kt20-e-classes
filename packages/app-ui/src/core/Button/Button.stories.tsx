import React from "react";
// import { action } from '@storybook/addon-actions';

import { Button } from "./Button.component";
import { Icon } from "../Icon/Icon.component";
import theme from "../theme";

export default {
  title: "Design System/Button",
  component: Button,
};

export const Basic = (args: any) => <Button {...args}>Label</Button>;

export const All = () => (
  <>
    <Button theme={theme} appearance="primary">
      Primary
    </Button>
    <Button theme={theme} appearance="secondary">
      Secondary
    </Button>
    <Button theme={theme} appearance="tertiary">
      Tertiary
    </Button>
    <Button theme={theme} appearance="outline">
      Outline
    </Button>
    <Button theme={theme} appearance="primaryOutline">
      Outline primary
    </Button>
    <Button theme={theme} appearance="secondaryOutline">
      Outline secondary
    </Button>
  </>
);

export const Sizes = () => (
  <>
    <Button theme={theme} appearance="primary">
      Default
    </Button>
    <Button theme={theme} appearance="primary" size="small">
      Small
    </Button>
  </>
);

export const Loading = () => (
  <>
    <Button theme={theme} appearance="primary" isLoading>
      Primary
    </Button>
    <Button theme={theme} appearance="secondary" isLoading>
      Secondary
    </Button>
    <Button theme={theme} appearance="tertiary" isLoading>
      Tertiary
    </Button>
    <Button theme={theme} appearance="outline" isLoading>
      Outline
    </Button>
    <Button
      theme={theme}
      appearance="outline"
      isLoading
      loadingText="Custom..."
    >
      Outline
    </Button>
  </>
);

export const Disabled = () => (
  <>
    <Button theme={theme} appearance="primary" disabled>
      Primary
    </Button>
    <Button theme={theme} appearance="secondary" disabled>
      Secondary
    </Button>
    <Button theme={theme} appearance="tertiary" disabled>
      Tertiary
    </Button>
    <Button theme={theme} appearance="outline" disabled>
      Outline
    </Button>
  </>
);

export const ContainsIcon = () => (
  <>
    <Button theme={theme} appearance="outline" containsIcon>
      <Icon icon="link" aria-label="Link" />
    </Button>
    <Button theme={theme} appearance="outline" size="small" containsIcon>
      <Icon icon="link" aria-label="Link" />
    </Button>
  </>
);
