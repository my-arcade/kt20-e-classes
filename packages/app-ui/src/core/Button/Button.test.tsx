import React from "react";
import { shallow, mount } from "enzyme";
import 'jest-styled-components';
import renderer from "react-test-renderer";
import Button from "./Button.component";
import theme from "../theme";

describe("Button Component", () => {
  it("should display the Button correctly", () => {
    const button = shallow(<Button theme={theme} />);
    expect(button).toMatchSnapshot();
  });

  it('should call an onclick function when clicked', () => {
    const mockFn = jest.fn();
    const wrapper = mount(<Button theme={theme} onClick={mockFn}/>);

    expect(mockFn).not.toHaveBeenCalled();

    wrapper.simulate('click');
    expect(mockFn).toHaveBeenCalled();

  })

  it("should display default button with correct background color", () => {
    const wrapper = renderer.create(<Button theme={theme} containsIcon/>).toJSON();

    expect(wrapper).toHaveStyleRule("background", 'transparent');
  });

  it("should display secondary button with correct background color ", () => {
    const wrapper = renderer.create(<Button theme={theme} appearance="secondary"/>).toJSON();

    expect(wrapper).toHaveStyleRule("background", theme.colors.secondary);
  });
});
