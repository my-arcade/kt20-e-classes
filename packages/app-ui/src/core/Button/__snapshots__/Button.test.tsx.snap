// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should display the Button correctly 1`] = `
<styled.button
  as={[Function]}
  theme={
    Object {
      "colors": Object {
        "background": "#FAFBFD",
        "border": "rgba(0,0,0,.1)",
        "clearBackground": "#FFFFFF",
        "danger": "#FF726C",
        "dark": "#666666",
        "darker": "#444444",
        "darkest": "#333333",
        "light": "#F3F3F3",
        "lighter": "#F8F8F8",
        "lightest": "#FFFFFF",
        "medium": "#DDDDDD",
        "mediumdark": "#999999",
        "mediumlight": "#EEEEEE",
        "metaPrimary": "#FFFBEB",
        "metaText": "rgb(42, 58, 95, 0.5)",
        "primary": "#FFC926",
        "secondary": "#2A3A5F",
        "success": "#58B606",
        "tertiary": "#919AAE",
        "text": "#2A3A5F",
        "warning": "#E69D00",
      },
      "spacing": Object {
        "borderRadius": Object {
          "default": 10,
          "small": 5,
        },
        "padding": Object {
          "large": 30,
          "medium": 20,
          "small": 10,
        },
      },
      "typography": Object {
        "size": Object {
          "code": 90,
          "l1": 32,
          "l2": 40,
          "l3": 48,
          "m1": 20,
          "m2": 24,
          "m3": 28,
          "s1": 12,
          "s2": 14,
          "s3": 16,
        },
        "type": Object {
          "code": "\\"SFMono-Regular\\", Consolas, \\"Liberation Mono\\", Menlo, Courier, monospace",
          "primary": "\\"Nunito Sans\\", \\"Mandali\\", Helvetica, Arial, sans-serif",
        },
        "weight": Object {
          "black": "900",
          "bold": "700",
          "extrabold": "800",
          "extralight": "200",
          "light": "300",
          "regular": "400",
        },
      },
    }
  }
>
  <styled.span />
</styled.button>
`;
